nixos/modules/tasks/network-interfaces.nix

```diff
@@ -541,6 +544,25 @@ in
       '';
     };
 
+    networking.machineId = mkOption {
+      default = null;
+      example = "b23d9f80c4494c00b632f2f6448aaa72";
+      type = types.nullOr types.str;
+      description = lib.mdDoc ''
+        The 128-bit machine ID of the machine, formatted as 32 hexadecimal characters.
```

Comment:
Maybe link to https://www.freedesktop.org/software/systemd/man/latest/machine-id.html